optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/BenchmarkResultIO.java

```diff
@@ -33,36 +33,24 @@
 import org.optaplanner.benchmark.impl.statistic.ProblemStatistic;
 import org.optaplanner.benchmark.impl.statistic.PureSubSingleStatistic;
 import org.optaplanner.core.config.solver.SolverConfig;
-import org.optaplanner.core.impl.solver.io.XStreamConfigReader;
-import org.optaplanner.persistence.xstream.api.score.AbstractScoreXStreamConverter;
+import org.optaplanner.core.impl.io.XmlUnmarshallingException;
+import org.optaplanner.core.impl.io.jaxb.JaxbIO;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.thoughtworks.xstream.XStream;
-import com.thoughtworks.xstream.XStreamException;
-import com.thoughtworks.xstream.converters.ConversionException;
-
 public class BenchmarkResultIO {
 
     private static final String PLANNER_BENCHMARK_RESULT_FILENAME = "plannerBenchmarkResult.xml";
 
     protected final transient Logger logger = LoggerFactory.getLogger(getClass());
 
-    private final XStream xStream;
-
-    public BenchmarkResultIO() {
-        xStream = XStreamConfigReader.buildXStream();
-        xStream.processAnnotations(PlannerBenchmarkResult.class);
-        xStream.allowTypesByRegExp(new String[] { "org\\.optaplanner\\.\\w+\\.api\\..*" });
-        xStream.allowTypesByRegExp(new String[] { "org\\.optaplanner\\.\\w+\\.impl\\..*" });
```

Comment:
(just thinking aloud, no action needed)

This is basically saying that all classes within the org.optaplanner namespace aren't exploitable.
That included org.optaplanner.examples, which did has this code in the dinner party example:

```
  <customPhase>
    <customPhaseCommandClass>org.optaplanner.examples.dinnerparty.solver.solution.initializer.DinnerPartySolutionInitializer</customPhaseCommandClass>
  </customPhase>
```

That has test coverage, so all is well with JAXB, I presume?

Reply:
JAXB accepts classes you specifically define when creating the JAXBContext. These are eligible to act as root elements. Then, it accepts any type it has to serialize in the object graph under these root element types.

This particular example is not about `DinnerPartySolutionInitializer`, though. The type to be (de)serialized here is `Class`.